.github/workflows/update-artifacts.yml

```diff
@@ -0,0 +1,54 @@
+name:  Update artifacts manifest
+
+on:
+  push:
+    branches:
+      -  main
+
+jobs:
+    update-artifacts:
+        runs-on: ubuntu-latest
+        steps:
+        - name: Checkout Code
+          uses: actions/checkout@v3
+          with:
+            fetch-depth: 0
+
+        - name: Setup Kubectl
+          id: install-kubectl
+          uses: azure/setup-kubectl@v3
+          with:
+            version: 'latest'
+
+        - name: Setup Helm
+          uses: azure/setup-helm@v4.2.0
+          with:
+            version: v3.15.1
+
+        - uses: actions/setup-python@v5
+          with:
+            python-version: '3.10'
+
+        - name: Install python dependencies
+          run: pip install -r scripts/artifacts/requirements.txt
+
+        # Generate artifacts manifest for inframold charts
+        - name: Generate Artifacts Manifest for Each Chart
+          run: |
+            charts_list=$(ls charts | grep inframold)
+            for chart in $charts_list;
+            do
+              version=$(cat charts/$chart/Chart.yaml | grep version | awk '{print $2}')
+              python scripts/artifacts/artifacts_template_generator.py $chart https://truefoundry.github.io/infra-charts/ \
+                $version charts/$chart/values.yaml charts/$chart/artifacts-manifest.json scripts/artifacts/extra.json
+            done
+
+        # Update the inframold artifacts manifest
+        - name: Create Pull Request
+          uses: peter-evans/create-pull-request@v6
+          with:
+            commit-message: Update Inframold artifacts manifest
+            title: [CI] Update Inframold Artifacts
+            body: Update Inframold artifacts manifest
+            branch: update-inframold-artifacts-manifest
```

Comment:
lets add a unique number to the title and branch to prevent reusing an older one if two PRs are raised simultaneously